### PR TITLE
[deps] Update functional dependencies

### PR DESCRIPTION
**Pull request by grimreaper**
Tests still pass

### CODE REVIEW

**Comment by LegacyKing** [comment]
Conflicts

**Comment by LegacyKing** [comment]
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/cdom/helper/AllowUtilities.java:21: error: package org.springframework.web.util does not exist
import org.springframework.web.util.HtmlUtils;
                                   ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/cdom/helper/AllowUtilities.java:83: error: cannot find symbol
				sb.append(HtmlUtils.htmlEscape(tempBuffer.toString()));
				          ^
  symbol:   variable HtmlUtils
  location: class AllowUtilities
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
 FAILED
:autoLintGradle SKIPPED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

**Comment by LegacyKing** [comment]
Still failing:

:compileJava/home/travis/build/PCGen/pcgen/code/src/java/pcgen/cdom/helper/AllowUtilities.java:21: error: package org.springframework.web.util does not exist
import org.springframework.web.util.HtmlUtils;
                                   ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/cdom/helper/AllowUtilities.java:83: error: cannot find symbol
				sb.append(HtmlUtils.htmlEscape(tempBuffer.toString()));
				          ^
  symbol:   variable HtmlUtils
  location: class AllowUtilities
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
 FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

**Comment by tildar** [comment]
for what it's worth, I've been using:

dependencies {
    compile group: 'cobra', name: 'cobra', version:'0.98.4-pcgen'
    compile group: 'commons-io', name: 'commons-io', version:'2.6'
    compile group: 'org.springframework', name: 'spring-core', version:'5.0.6.RELEASE'
    compile group: 'org.springframework', name: 'spring-beans', version:'5.0.6.RELEASE'
    compile group: 'org.springframework', name: 'spring-web', version:'5.0.6.RELEASE'
    compile group: 'skinlf', name: 'skinlf', version: '1.2.3'
    compile group: 'org.apache.commons', name: 'commons-lang3', version:'3.7'
    compile group: 'org.apache.xmlgraphics', name: 'fop', version:'2.3'
    compile group: 'org.apache.commons', name: 'commons-collections4', version: '4.1'
    compile group: 'jep', name: 'jep', version:'2.4.1'
    compile group: 'org.freemarker', name: 'freemarker', version:'2.3.28'
    compile group: 'org.jdom', name: 'jdom2', version:'2.0.6'
    compile group: 'xalan', name: 'xalan', version:'2.7.2'
    compile group: 'net.sourceforge.argparse4j', name: 'argparse4j', version: '0.8.1'
    compile group: 'org.mozilla', name: 'rhino', version:'1.7.10'

**Comment by LegacyKing** [comment]
@karianna calling this a dead PR. No action from the OP for two months.

File: build.gradle [hunk 1]
```diff
@@ -161,19 +161,18 @@ compileJava {
 dependencies {
     compile group: 'cobra', name: 'cobra', version:'0.98.4-pcgen'
     compile group: 'commons-io', name: 'commons-io', version:'2.6'
-    compile group: 'org.springframework', name: 'spring-core', version:'5.0.3.RELEASE'
-    compile group: 'org.springframework', name: 'spring-beans', version:'5.0.3.RELEASE'
-    compile group: 'org.springframework', name: 'spring-web', version:'5.0.3.RELEASE'
+    compile group: 'org.springframework', name: 'spring-core', version:'5.0.5.RELEASE'
+    compile group: 'org.springframework', name: 'spring-beans', version:'5.0.5.RELEASE'
```
Comment (tildar) [inline build.gradle]: compile group: 'org.springframework', name: 'spring-web', version:'5.0.5.RELEASE'

Comment (karianna) [reply build.gradle]: I've added it directly, lets see if this passes

**Comment by tildar** [comment]
@karianna 
Let's drop this one. It would  probably builds fine if travis was restarted, but  I have a more updated version in #4118 anyway.

**Pull request closed without merging**